tokio/src/blocking.rs

```diff
@@ -0,0 +1,41 @@
+cfg_rt_core! {
+    pub(crate) use crate::runtime::spawn_blocking;
+    pub(crate) use crate::task::JoinHandle;
+}
+
+cfg_not_rt_core! {
+    use std::fmt;
+    use std::future::Future;
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+    pub(crate) fn spawn_blocking<F, R>(_f: F) -> JoinHandle<R>
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static,
+    {
+        panic!("requires the `rt-core` Tokio feature flag")
+
+    }
+
+    pub(crate) struct JoinHandle<R> {
+        _p: std::marker::PhantomData<R>,
+    }
```

Comment:
The `Sync` impl of this dummy `JoinHandle` doesn't match the real `JoinHandle`. This could cause code that compiles with just `blocking` to fail when `rt-core` is added.

You could add the `spawn` function to [`async_send_sync.rs`][1] to verify that this is correct.

[1]: https://github.com/tokio-rs/tokio/blob/master/tokio/tests/async_send_sync.rs

Reply:
That file only runs w/ all feature flags. I added a test in `blocking.rs` itself.